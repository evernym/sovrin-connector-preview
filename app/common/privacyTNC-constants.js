// @flow

export { EULA_URL as TermsAndConditionUrl } from '../eula/type-eula'
export const TermsAndConditionsTitle = 'Terms and Conditions'

export const PrivacyPolicyUrl = 'https://www.connect.me/privacy.html'
export const PrivacyPolicyTitle = 'Privacy Policy'
