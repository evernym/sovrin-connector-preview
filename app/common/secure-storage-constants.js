// @flow

export const PUSH_COM_METHOD = 'pushComMethod'
export const CONNECT_ME_STORAGE_KEY = '@sovrinconnectorDefaults'
export const IS_ALREADY_INSTALLED = `${CONNECT_ME_STORAGE_KEY}:IS_ALREADY_INSTALLED`
export const CONNECTIONS = 'CONNECTIONS'
export const CLAIM_MAP = `${CONNECT_ME_STORAGE_KEY}:'CLAIM_MAP`
export const WALLET_ENCRYPTION_KEY = 'walletEncryptionKey'
export const WALLET_BALANCE = 'walletBalance'
export const WALLET_ADDRESSES = 'walletAddresses'
export const WALLET_HISTORY = 'walletHistory'
export const LAST_SUCCESSFUL_BACKUP = 'lastSuccessfulBackup'
export const PASSPHRASE_SALT_STORAGE_KEY = 'passphraseSalt'
export const PASSPHRASE_STORAGE_KEY = 'passphrase'
