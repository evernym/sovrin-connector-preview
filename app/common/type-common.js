// @flow

import type { Element } from 'react'

export type CustomError = {
  code: string,
  message: string,
}

export const INITIAL_TEST_ACTION: 'INITIAL_TEST_ACTION' = 'INITIAL_TEST_ACTION'

export type InitialTestAction = {
  type: typeof INITIAL_TEST_ACTION,
}

export const initialTestAction = () => ({
  type: INITIAL_TEST_ACTION,
})

export type NavigationParams = {
  [string]: any,
}

export type ReactNavigation = {
  navigation: {
    navigate: (route: string, params?: any) => void,
    state?: {
      // TODO:KS This has to accept a Generic and every where we are using
      // react navigation params, we have to pass a generic for type safety
      params: NavigationParams,
    },
    goBack: (route?: ?string) => void,
    // TODO: Had to use any here because we don't know type of beforehand
    // need to move react navigation library to latest version
    // and use their type, also need to use Generics here
    dispatch: any => void,
    setParams: (params?: any) => void,
    isFocused: boolean,
  },
}

export type GenericObject = {
  [string]: any,
}

export type GenericStringObject = {
  [string]: string,
}

export type MessageAnnotation = {
  name: string,
  version: string,
}

export type TopicAnnotation = {
  tid: number,
  mid: number,
}

export type ReactChildren = Element<*>

export const RESET: 'RESET' = 'RESET'
export const REMOVE_SERIALIZED_CLAIM_OFFERS_SUCCESS: 'REMOVE_SERIALIZED_CLAIM_OFFERS_SUCCESS' =
  'REMOVE_SERIALIZED_CLAIM_OFFERS_SUCCESS'

export type ResetAction = {
  type: typeof RESET,
}

export type ImageSource = {
  uri: string,
}

export type NotificationPayload = {
  forDID: string,
  uid: string,
  type: string,
  remotePairwiseDID: string,
  senderLogoUrl?: ?string,
}

export type ImmutableGenericStringObject = {
  +[string]: string,
}
