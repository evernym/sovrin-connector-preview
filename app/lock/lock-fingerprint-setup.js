// @flow
import React, { PureComponent } from 'react'
import { Alert, StyleSheet } from 'react-native'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { TouchId } from '../components/touch-id/touch-id'
import { Container } from '../components'
import { captureError } from '../services/error/error-handler'
import type { Store } from '../store/type-store'

import {
  lockPinSetupRoute,
  lockSelectionRoute,
  settingsTabRoute,
} from '../common'

import {
  OFFSET_2X,
  OFFSET_3X,
  OFFSET_6X,
  isiPhone5,
} from '../common/styles/constant'
import { disableTouchIdAction, enableTouchIdAction } from '../lock/lock-store'
import { AllowedFallbackToucheIDErrors } from './type-lock'
import type { LockFingerprintSetupProps } from './type-lock'

export class LockFingerprintSetup extends PureComponent<
  LockFingerprintSetupProps,
  void
> {
  goToSettingsScreen = () => {
    if (this.props.touchIdActive) {
      this.props.disableTouchIdAction()
      Alert.alert(
        `You'll need to use your pass code to unlock this app from now on`,
        null,
        [
          {
            text: 'OK',
            onPress: () => this.props.navigation.goBack(null),
          },
        ],
        { cancelable: false }
      )
    } else {
      this.props.enableTouchIdAction()
      this.props.navigation.goBack(null)
    }
  }

  goToPinSetupScreen = () => {
    this.props.enableTouchIdAction()
    this.props.navigation.navigate(lockPinSetupRoute, { touchIdActive: true })
  }

  touchIdHandler = () => {
    TouchId.isSupported()
      .then(success => {
        TouchId.authenticate('', this.touchIdHandler)
          .then(success => {
            this.props.fromSettings
              ? this.goToSettingsScreen()
              : this.goToPinSetupScreen()
          })
          .catch(error => {
            captureError(error)
            if (AllowedFallbackToucheIDErrors.indexOf(error.name) >= 0) {
              this.props.fromSettings
                ? this.props.navigation.goBack(null)
                : this.props.navigation.navigate(lockSelectionRoute)
            }
          })
      })
      .catch(error => {
        captureError(error)
        if (AllowedFallbackToucheIDErrors.indexOf(error.name) >= 0) {
          this.props.fromSettings
            ? this.props.navigation.goBack(null)
            : this.props.navigation.navigate(lockSelectionRoute)
        }
      })
  }

  componentDidMount() {
    this.touchIdHandler()
  }

  render() {
    return <Container tertiary style={[style.pinSelectionContainer]} />
  }
}

const mapStateToProps = (state: Store, props) => ({
  touchIdActive: state.lock.isTouchIdEnabled,
  fromSettings:
    props.navigation.state.params !== undefined
      ? props.navigation.state.params.fromSettings
      : false,
})

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      disableTouchIdAction,
      enableTouchIdAction,
    },
    dispatch
  )

const style = StyleSheet.create({
  pinSelectionContainer: {
    paddingTop: OFFSET_3X,
    paddingBottom: isiPhone5 ? OFFSET_2X : OFFSET_6X,
    paddingHorizontal: OFFSET_2X,
  },
})

export default connect(mapStateToProps, mapDispatchToProps)(
  LockFingerprintSetup
)
