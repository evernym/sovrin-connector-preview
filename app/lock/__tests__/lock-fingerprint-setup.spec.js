// @flow
import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { PIN_SETUP_STATE } from '../type-lock'
import { LockFingerprintSetup } from '../lock-fingerprint-setup'
import { lockPinSetupRoute } from '../../common'
import { getNavigation } from '../../../__mocks__/static-data'

describe('<LockFingerprintSetup />', () => {
  const getProps = () => ({
    navigation: getNavigation({ touchIdActive: true }),
    touchIdActive: true,
    fromSettings: false,
    disableTouchIdAction: jest.fn(),
    enableTouchIdAction: jest.fn(),
  })

  const options = {
    createNodeMock: element => {
      return {
        clear: () => {
          cleared = true
        },
      }
    },
  }

  let component
  let props
  let cleared

  beforeEach(() => {
    props = getProps()
    component = renderer.create(<LockFingerprintSetup {...props} />, options)
  })

  it('should render touchId modal', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should test touchId handler is called', () => {
    let tree = component.toJSON()
    let instance = component.getInstance()
    //manually set to true"
    instance.touchIdHandler()
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
