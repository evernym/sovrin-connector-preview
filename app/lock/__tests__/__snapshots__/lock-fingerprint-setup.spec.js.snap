// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LockFingerprintSetup /> should render touchId modal 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "paddingBottom": 20,
      "paddingHorizontal": 20,
      "paddingTop": 30,
    }
  }
  tertiary={true}
/>
`;

exports[`<LockFingerprintSetup /> should test touchId handler is called 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "paddingBottom": 20,
      "paddingHorizontal": 20,
      "paddingTop": 30,
    }
  }
  tertiary={true}
/>
`;
